Modules/rat_facts.py

```diff
@@ -0,0 +1,187 @@
+        else:
+            timestamp = datetime.datetime.utcfromtimestamp(0)
+        if result:
+            result = result[0]
+            return Fact(
+                result[0],
+                result[1],
+                result[2],
+                result[3],
+                timestamp
+            )
+        else:
+            return None
+
+    async def is_fact(self, name: str, lang: str = "en") -> bool or None:
+        """
+        checks for existence of a fact
+        Args:
+            name: name of the fact to check
+            lang: language to check
+
+        Returns: True if it exists, False if it exists
+
+        """
+        result = await self.dbm.select_rows("fact", "AND", {"lang": lang, "name": name})
+        return len(result) > 0
+
+    async def set_fact(self, fact: Fact) -> bool or None:
+        """
+        sets a fact
+        Args:
+            fact (Fact): the fact to push into the DB
+        Returns: True on success, False if it failed
+
+        """
+        try:
+            await self.dbm.insert_row("fact",
+                                      {"name": fact.name,
+                                       "lang": fact.lang,
+                                       "message": fact.message,
+                                       "author": fact.author},
+                                      ("name", "lang",)
+                                      )
+        except ValueError:
+            return False
+        else:
+            time = datetime.datetime.now(tz=datetime.timezone.utc)
+            self.dbm._execute(r"INSERT INTO fact_timestamps (name, lang, last_modified) "
+                              r"VALUES("
+                              f"'{fact.name}', '{fact.lang}', ?)"
+                              r"ON CONFLICT (name, lang)"
+                              r"DO UPDATE SET last_modified = ?",
+                              time, time  # time is doubled on purpose
+                              )
+            return True
+
+    async def delete_fact(self, name: str, lang: str) -> bool or None:
+        """
+        delete's the given fact
+        Args:
+            name: name of the fact to delete
+            lang: language to delete
+
+        Returns: True on success, False if it failed and None if the DBM is disabled
+
+        """
+        if not self.dbm.enabled:
+            return None
+        try:
+            await self.dbm.delete_row("fact", "AND", {"name": name, "lang": lang})
+            return True
+        except ValueError:
+            return False
+
+    async def handle_fact(self, context: Context) -> bool:
+        """
+
+        Responds with the message of the fact, should a match be found and not listed as an
+        internal fact.
+
+        Args:
+            context: the context as passed by rat_command
+
+        Returns: True if it was a fact, False if not. Will also return False,
+            should the DBM be disabled.
+
```

Comment:
This docstring is confusing. the description section describes it as `Responds with the message of the fact`, which i interpret as a return.

Later in the `Returns` block it returns a boolean...

Reply:
amended to clarify